### PR TITLE
Added Basic Documentation for the CLI

### PR DESCRIPTION
**Pull request by nmn**
Added another section on the documentation page with a single page explaining how the CLI works and how to use it find out about its capabilities.

### CODE REVIEW

File: docs/04-start-server.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+permalink: /docs/cli.html
+prev: cli.html
+---
+
+The flow command line tool is made to be easy-to-use for the simple case.
+
+Just using the command `flow` will type-check your current directory if the `.flowconfig` file is present.
+A flow server will automatically be started if needed.
+
+However, the cli tool gives you a more control if you want, and also has the ability to provide tools to build your own IDE plug-ins which work like [Nuclide](http://nuclide.io).
```
Comment (jeffmo) [inline docs/04-start-server.md]: Bit of a run-on sentence here, it seems. Maybe just:

```
The CLI tool also provides several other options and commands that allow you to control the server and build tools that integrate with Flow. For example, this is how the [Nuclide](http://nuclide.io) editor integrates with Flow to provide autocompletion, type errors, etc in it's UI.

To find out more [...]
```

File: docs/04-start-server.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+id: cli
+title: Using the CLI
+layout: docs
+permalink: /docs/cli.html
+prev: cli.html
+---
+
+The flow command line tool is made to be easy-to-use for the simple case.
+
+Just using the command `flow` will type-check your current directory if the `.flowconfig` file is present.
```
Comment (jeffmo) [inline docs/04-start-server.md]: Woohoo! It seems like many people assume `flow check` to be the default when really `flow` should be the common case. Hopefully these docs will help clarify this for everyone.

File: index.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<!DOCTYPE html>
```
Comment (jeffmo) [inline index.html]: Hmm, I think this file (and the ones below it) are build artifacts and aren't supposed to be checked in...

**Comment by jeffmo** [comment]
Thank you so much for writing this! One step closer to better docs :)

**Comment by nmn** [comment]
Anything you need me to do to get this merged in? I can make some edits to the text as per your comments?

**Comment by jeffmo** [comment]
If you can remove the build artifacts and tweak that one sentence then I can merge this for you

**Comment by nmn** [comment]
OK will do.

**Comment by nmn** [comment]
Done and Done.

**Comment by jeffmo** [comment]
You're awesome!

**Pull request merged**